### ISSUE TITLE
Create QA process + metrics for Line Extraction

### ISSUE THREAD

**Issue by jarmoza** [printprobability/qa-workflow#9]
Create a new module/function for the qa script that runs line extraction or looks at already processed line extraction and calculates the following metrics.

**Metrics**
1. area of page that is lines
2. variance of line height

**Comment by jarmoza** [comment]
**Notes**

put autocrop output in separate directory for line extraction
run line extraction prepprocessing in run_workflow1_watershed.sh
line_extract_dhsegment.sh is run from run_workflow1_watershed.sh after workflow0 and post-autocrop processing

Metrics 
num lines on page
median line height in pixels in page
variance line height in pixels on page

1. num lines on page
2. median line height in pixels in page
3. variance line height in pixels on page

Calculate stuff over the pages in post

Line extraction is further down in run_workflow0.sh

**Comment by jarmoza** [comment]
**run_workflow1_watershed description**

**Stage 1 - Preprocessing for Line Extraction**

1. Confirm we have a directory with enumerated tifs
2. Convert color tifs to grayscale
3. resized directory appears with permissions rwx
4. tif files are copied to resize
5. moving tif files to an originals folder
6. Change to resized directory
6. localthresh run on tif files here
7. tifs files in resized permissions changed to rwx
8. tifs files in new gray2 directory permissions changed to rwx
9. Check that grayscale and color images are same size
10. Move tifs in resized to new 'book_color' directory
11. Move tifs in gray2 to current directory (resized)
12. Change all group file permissions in the book directory to rwx

**Stage 2/3: B&W / Color Line Extraction**

1. Switch to the top level book directory
2. **Run line_extract_dhsegment.dh**
  A. Date is shown
  B. Enables Max's conda environment at gsell/.conda/envs/my_env
  C. Make 'book' and 'book_color' subdirectories
  D. Copy original images to 'book_color' subdirectory
  E. Move page tifs to 'book' subdirectory
  F. Make directory pages
  G. Make directory pages_color
  H. Copy all tifs in 'book' subdirectory to 'pages'
  I. Copy all tifs in book_color to pages_color
  J. Copy code/run_dhsegment_on_book.py to 'pages'
  K. Deactivate Max's conda environment
  L. Activate Nikolai's environment at nikolaiv/miniconda3/envs/dh_segment
  M. Make lines directory
  N. Make lines_color directory
  O. Change directory to pages
  P. **Run run_dhsegment_on_book.py (recently copied here in 'pages')**
  Q. Move back up one directory (cd ..)
  R. Copy code/do_all_lines_dhsegment.py lines
  S. Change directory to lines
  T. Copy code/watershed_line_extraction.py to lines (present directory)
  U. **Run watershed_line_extraction.py (recently copied here in 'lines') with the following arguments**
    i. -u for python3 (unbuffered output) - so 'python3 -u watershed_line_extraction.py'
    ii. ../pages_color
    iii. ../pages
    iv. ..pages/dhsegment_output (first time I'm seeing this dhsegment_output subdirectory?)
    v. --lines_output_directory .
    vi. --color_lines_output_directory
    vii. ../lines_color
    viii. --max_height 200
    ix. --min_width 200
    x. --extension .tif
    xi. --line_height_quantile 0.85
    xii. --transformations_csv
    xiii. ../lines_color/transformations.csv
  V. Date is shown
  W. Move back up on directory (cd ..)
  X. Copy original images back to top-level book directory?
3. If line_extract_dhsegment.sh (and its processes) complete successfully, continue
4. Generate a montage of the tif files in the line directory into a single tif 'lines_to_check.tif' (likely can be skipped since we're developing an automated QA process.)
5. Convert 'lines_to_check.tif' to 'lines_to_check.jpg'
6. Continue only if lines_color/transformations.csv exists
7. Change group permissions for all files in the top level book directory to rwx
8. Note that https://printprobdb.psc.edu/iiif/<book directory in shared>/lines_to_check.jpg is viewable by browser

**Comment by jarmoza** [comment]
NOTE: B&W images must be created for watershed_line_extraction to work

**Comment by jarmoza** [comment]
Blocked by https://github.com/printprobability/qa-workflow/issues/12

**Comment by jarmoza** [comment]
QA for line extraction including `run`, `output_stats`, and `collate` are functional. There is still some cleanup and the need to refactor/adapt the code when an alternate line extraction method is introduced, but I will open another issue for spec'ing that implementation out.
